src/coreclr/src/gc/gc.cpp

```diff
@@ -35843,7 +35645,7 @@ gc_heap::verify_free_lists ()
                                  (size_t)free_list));
                     FATAL_GC_ERROR();
                 }
-                if ((gen_num != max_generation+1)&&(object_gennum (free_list)!= gen_num))
+                if ((gen_num <= max_generation)&&(object_gennum (free_list)!= gen_num))
```

Comment:
nit - would be good to take the opportunity to fix the spacing here -

`if ((gen_num <= max_generation) && (object_gennum (free_list)!= gen_num))`